### PR TITLE
Fixing error where lock could not be taken on build folder

### PR DESCRIPTION
**Pull request by sokket**
cc @piotrpMSFT 

Ran this and it was able to take the lock, this should fix the 404 blob not found problem

---
This change is [Reviewable](https://reviewable.io/reviews/dotnet/cli/2774)

### CODE REVIEW

**Comment by sokket** [comment]
@dotnet-bot test this please

**Comment by sokket** [comment]
@dotnet-bot test  OSX x64 Release Build

File: scripts/dotnet-cli-build/PublishTargets.cs [hunk 1]
```diff
@@ -60,7 +60,9 @@ public static BuildTargetResult FinalizeBuild(BuildTargetContext c)
             {
                 string targetContainer = $"{Channel}/Binaries/Latest/";
                 string targetVersionFile = $"{targetContainer}{CliNuGetVersion}";
-                string leaseId = AzurePublisherTool.AcquireLeaseOnBlob(targetContainer);
+                string semaphoreBlob = $"{Channel}/Binaries/publishSemaphore";
```
Comment (Sridhar-MS) [inline scripts/dotnet-cli-build/PublishTargets.cs]: Should we have something similar for `{Channel}/Installers/publishSemaphore`

Comment (sokket) [inline scripts/dotnet-cli-build/PublishTargets.cs]: That is published in a different code-path, is it not?

Comment (Sridhar-MS) [inline scripts/dotnet-cli-build/PublishTargets.cs]: Are we not 'Finalizing' installer artifacts too..

Comment (sokket) [inline scripts/dotnet-cli-build/PublishTargets.cs]: I don't believe in this code path; we should only be touching the binaries/Latest folder here. Adding a semaphore for the installer binaries would be out-of-scope for this PR

**Comment by sokket** [comment]
@dotnet-bot test this please

**Comment by sokket** [comment]
@dotnet-bot test Windows_NT x64 Release Build

**Comment by sokket** [comment]
Any comments on this or is it good to go when green?

**Comment by piotrpMSFT** [comment]
:shipit:

**Comment by sokket** [comment]
Thanks! Merging

**Pull request merged**